packages/authentication/src/__tests__/acceptance/basic-auth-extension.acceptance.ts

```diff
@@ -0,0 +1,199 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/authentication
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {inject} from '@loopback/context';
+import {addExtension, Application} from '@loopback/core';
+import {anOpenApiSpec} from '@loopback/openapi-spec-builder';
+import {api, get} from '@loopback/openapi-v3';
+import {RestServer} from '@loopback/rest';
+import {Client, createClientForHandler} from '@loopback/testlab';
+import {authenticate, AuthenticationBindings, UserProfile} from '../..';
+import {getApp, getUserRepository} from '../fixtures/helper';
+import {BasicAuthenticationStrategyBindings, USER_REPO} from '../fixtures/keys';
+import {MyAuthenticationSequence} from '../fixtures/sequences/authentication.sequence';
+import {BasicAuthenticationUserService} from '../fixtures/services/basic-auth-user-service';
+import {BasicAuthenticationStrategy} from '../fixtures/strategies/basic-strategy';
+import {UserRepository} from '../fixtures/users/user.repository';
+
+describe('Basic Authentication', () => {
+  let app: Application;
+  let server: RestServer;
+  let users: UserRepository;
+  beforeEach(givenAServer);
+  beforeEach(givenControllerInApp);
+  beforeEach(givenAuthenticatedSequence);
+  beforeEach(givenProviders);
+
+  it(`authenticates successfully for correct credentials for user 'jack'`, async () => {
+    const client = whenIMakeRequestTo(server);
+    const credential =
+      users.list['joe@example.com'].user.email +
+      ':' +
+      users.list['joe@example.com'].user.password;
+    const hash = Buffer.from(credential).toString('base64');
+    await client
+      .get('/whoAmI')
+      .set('Authorization', 'Basic ' + hash)
```

Comment:
Please refactor the logic to build `Authorization` header from user/pass into a helper function and reuse it for all tests.

Reply:
I will create a utility to base64 encode `email:password` or any combination of strings required for my tests (valid and invalid combinations)

e.g.
```
.set('Authorization', 'Basic ' + createBase64Hash(users.list['joe@example.com'].email, users.list['joe@example.com'].password))
or 
.set('Authorization', 'NotB@s1c ' + createBase64Hash(users.list['joe@example.com'].email, users.list['joe@example.com'].password))
or 
.set('Authorization', 'Basic ' + createBase64Hash(users.list['joe@example.com'].email, users.list['joe@example.com'].password, {alternativeSeparator: '|'}))
or
.set('Authorization', 'Basic ' + createBase64Hash(users.list['joe@example.com'].email, users.list['joe@example.com'].password, {extraSegment: 'extraPart'}))
```